src/views/ManageList.jsx

```diff
@@ -17,9 +17,34 @@ export function ManageList({ listPath, userId }) {
 		setShareEmail(target.value);
 	};
 
+	const currentList = data.map((item) => {
+		return item.name
+			.toLowerCase()
+			.replace(/[!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~/\s]/g, '');
+	});
+
 	const handleItemSubmit = async (event) => {
 		newItem.daysUntilNextPurchase = Number(newItem.daysUntilNextPurchase);
 		event.preventDefault();
+
+		//edge case: what if someone decides to put only puncutation as item name
+		if (newItem.itemName) {
+			const itemName = newItem.itemName
+				.toLowerCase()
+				.replace(/[!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~/\s]/g, '');
+
+			if (currentList.includes(itemName)) {
+				alert('This item already exists in the list.');
+				return;
+			}
+		}
+
+		if (!newItem.itemName) {
+			alert('Please enter an item name');
+			return;
+		}
```

Comment:
refactor: we could move the second if clause to be early, so we don't need to have nested if for the first if clause

```suggestion
		if (!newItem.itemName) {
			alert('Please enter an item name');
			return;
		}

		//edge case: what if someone decides to put only puncutation as item name
		const itemName = newItem.itemName
			.toLowerCase()
			.replace(/[!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~/\s]/g, '');
		console.log('itemName:', itemName);

		if (currentList.includes(itemName)) {
			alert('This item already exists in the list.');
			return;
		}
```